### PR TITLE
New `ReleaseVersion` component into `HelpButton`

### PR DESCRIPTION
**Pull request by hoshinotsuyoshi**
## Summary

Example output for cliVersion:
- Released version:
  v0.0.11 (2024-12-19)
- Unreleased version:
  v0.0.11 + 0d6169a (2024-12-19)

Explanation:
- "Released version" means the current Git hash matches a tagged release.

スクリーンショット 2024-12-19 9 12 34

- "Unreleased version" includes a short Git hash prefix to indicate changes since the last release.

スクリーンショット 2024-12-19 9 11 47

## Changes

* b400891e : feat(cli): inject new `VITE_CLI_VERSION_*` env vars into Vite build process
* 6fdbbdd3 : feat: add `CliVersionProvider` and schema definition for CLI version context
* 343e01d9 : :sparkles: New `ReleaseVersion` component into `HelpButton`

## Testing

Manually tested.

## Other Information

In pull requests, the commit hash displayed in the HelpButton may differ from the hash of the latest commit. This happens because when using `on: pull_request` in GitHub Actions, a special merge is performed before the code is checked out.

### CODE REVIEW

File: frontend/packages/cli/package.json [hunk 1]
```diff
@@ -37,11 +37,11 @@
   "scripts": {
     "build": "pnpm run '/^build:.*/'",
     "build:cli": "rollup -c  && pnpm run cp:prism",
-    "build:vite": "vite build",
+    "build:vite": "node scripts/set-env.mjs vite build",
     "command:build": "node ./dist-cli/bin/cli.js erd build --input fixtures/input.schema.rb --format schemarb",
     "cp:prism": "cp ../db-structure/node_modules/@ruby/prism/src/prism.wasm ./dist-cli/bin/prism.wasm",
     "dev": "pnpm command:build && cp dist/schema.json public/ && pnpm run '/^dev:.*/'",
-    "dev:app": "vite",
+    "dev:app": "node scripts/set-env.mjs vite",
```
Comment (hoshinotsuyoshi) [inline frontend/packages/cli/package.json:44]: `node scripts/set-env.mjs` was needed as well as "build:vite"

File: frontend/packages/erd-core/src/schemas/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export * from './cliVersion'
+export * from './queryParam'
+export * from './showMode'
```
Comment (hoshinotsuyoshi) [inline frontend/packages/erd-core/src/schemas/index.ts:3]: I only wanted the cliVersion here, but I'm also exporting the others for consistency

File: frontend/packages/erd-core/src/providers/index.ts
Comment (hoshinotsuyoshi) [inline frontend/packages/erd-core/src/providers/index.ts:1]: I created a new directory, but is it appropriate? 💭

**Review by MH4GF — APPROVED** [review]
Great!!
I've wrote some comments!

File: frontend/packages/erd-core/src/components/ERDRenderer/AppBar/HelpButton/ReleaseVersion.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    <div className={styles.cliVersion}>
+      <div className={styles.cliVersionInner}>
+        <span>{`v${cliVersion.version}`}</span>
+        <span>
+          {' '}
+          {cliVersion.isReleasedGitHash ||
+            `+ ${cliVersion.gitHash.slice(0, 7)} `}
+        </span>
+        <span>{cliVersion.date.length > 0 && ` (${cliVersion.date})`}</span>
+      </div>
```
Comment (hoshinotsuyoshi) [inline frontend/packages/erd-core/src/components/ERDRenderer/AppBar/HelpButton/ReleaseVersion.tsx:27]:
```suggestion
      <span>{`v${cliVersion.version}`}</span>
      <span>
        {' '}
        {cliVersion.isReleasedGitHash ||
          `+ ${cliVersion.gitHash.slice(0, 7)} `}
      </span>
      <span>{cliVersion.date.length > 0 && ` (${cliVersion.date})`}</span>
```

File: frontend/packages/erd-core/src/components/ERDRenderer/AppBar/HelpButton/ReleaseVersion.module.css [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+.cliVersionInner {
+  border-radius: var(--border-radius-full);
+  background: var(--pane-muted-background);
+  padding: var(--spacing-1) var(--spacing-2);
+  color: var(--global-mute-text);
+  font-family: var(--code-font);
+  font-size: var(--font-size-1);
+}
```
[the remark this answers is not in the document]
Comment (hoshinotsuyoshi) [reply frontend/packages/erd-core/src/components/ERDRenderer/AppBar/HelpButton/ReleaseVersion.module.css:12]: I see!

**Pull request merged**